### ISSUE TITLE
Database Error when Searching for Products in Admin

### ISSUE THREAD

**Issue by pauldowlin** [woocommerce/woocommerce#28672]
I was able to narrow it down to the use of MySQL 8. If I use MySQL 5.6, everything else being equal, then Product search's in Admin works as expected. Product Search was working at one point with our server/database setup (MySQL 8.0) but unfortunately I have procrastinated long enough that I don't have any setups running old enough versions of Woocommerce to nail down what version this changed on (somewhere around 4.5?).
And this problem only affects Product Searches (Product Edit page, Coupon creation, etc) . Any other searches in Admin work as expected.

Thanks for taking a look.

**Prerequisites (mark completed items with an [x]):**
- [ x ] I have have carried out troubleshooting steps and I believe I have found a bug.
- [ x ] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

**Describe the bug**
When searching for products on Admin Woocommerce / Product page a Database error occurs, Unknown column '' in 'where clause' and no products are returned for the search word/phrase used.

**Expected behavior**
No database error and matching products returned.

**Actual behavior**
No products are returned for the search terms entered even though they exist in product title, content, etc.

**Steps to reproduce the bug (We need to be able to reproduce the bug in order to fix it.)**
Steps to reproduce the bug:
1. See Wordpress Environment, specifically the use of MySQL 8 
2. Go to Woocommerce/Products in Wordpress Admin
3. Enter in product search terms in 'Search products' input box
4. Click on Search products button
5. See error
6. See My Resolution

**Error from Query monitor**
Query:
```
SELECT DISTINCT posts.ID as product_id, posts.post_parent as parent_id
FROM wp_f0d401a675_posts posts
LEFT JOIN wp_f0d401a675_wc_product_meta_lookup wc_product_meta_lookup
ON posts.ID = wc_product_meta_lookup.product_id
LEFT JOIN wp_f0d401a675_wc_product_meta_lookup parent_wc_product_meta_lookup
ON posts.post_type = 'product_variation'
AND parent_wc_product_meta_lookup.product_id = posts.post_parent
WHERE posts.post_type IN ('product','product_variation')
AND ( ( ( posts.post_title LIKE '%hat%')
OR ( posts.post_excerpt LIKE '%hat%')
OR ( posts.post_content LIKE '%hat%' )
OR ( wc_product_meta_lookup.sku LIKE '%hat%' )
OR ( wc_product_meta_lookup.sku = ""
AND parent_wc_product_meta_lookup.sku LIKE '%hat%' ) ))
ORDER BY posts.post_parent ASC, posts.post_title ASC
```
Caller:
`WC_Product_Data_Store_CPT->search_products()`

Error Message:
`Unknown column '' in 'where clause'`

Error Code:
`1054`

**My Resolution**
After much searching and debugging I was able to narrow it down to the following line in the Query used
`OR ( wc_product_meta_lookup.sku = ""`
When I replaced the double quotes with single quotes the query completed as expected.
`OR ( wc_product_meta_lookup.sku = ''`

**Isolating the problem (mark completed items with an [x]):**
- [ x ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ x ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [ x ] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
We use MySQL 8 on a Digital Ocean Managed Database

<details>
`
### WordPress Environment ###

WordPress address (URL): 
Site address (URL): 
WC Version: 4.6.2
REST API Version: ✔ 4.6.2
WC Blocks Version: ✔ 3.4.0
Action Scheduler Version: ✔ 3.1.6
WC Admin Version: ✔ 1.6.2
Log Directory Writable: ✔
WP Version: ❌ 5.5.3 - There is a newer version of WordPress available (5.6)
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.46 (Unix) OpenSSL/1.1.1
PHP Version: 7.4.13
PHP Post Max Size: 1 GB
PHP Time Limit: 30
PHP Max Input Vars: 5000
cURL Version: 7.58.0
OpenSSL/1.1.1

SUHOSIN Installed: –
MySQL Version: 8.0.19
Max Upload Size: 512 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 4.6.2
WC Database Prefix: wp_f0d401a675_
Total Database Size: 35.50MB
Database Data Size: 31.43MB
Database Index Size: 4.07MB
wp_f0d401a675_woocommerce_sessions: Data: 0.06MB + Index: 0.02MB + Engine InnoDB
wp_f0d401a675_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_f0d401a675_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_f0d401a675_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_f0d401a675_woocommerce_order_items: Data: 0.06MB + Index: 0.02MB + Engine InnoDB
wp_f0d401a675_woocommerce_order_itemmeta: Data: 0.30MB + Index: 0.30MB + Engine InnoDB
wp_f0d401a675_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_f0d401a675_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_f0d401a675_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_f0d401a675_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_f0d401a675_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_f0d401a675_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_f0d401a675_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_f0d401a675_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_f0d401a675_actionscheduler_actions: Data: 0.09MB + Index: 0.11MB + Engine InnoDB
wp_f0d401a675_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_f0d401a675_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_f0d401a675_actionscheduler_logs: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
wp_f0d401a675_atum_inventories: Data: 0.05MB + Index: 0.08MB + Engine InnoDB
wp_f0d401a675_atum_inventory_locations: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_f0d401a675_atum_inventory_meta: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_f0d401a675_atum_inventory_orders: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_f0d401a675_atum_inventory_regions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_f0d401a675_atum_inventory_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_f0d401a675_atum_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_f0d401a675_atum_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_f0d401a675_atum_product_data: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_f0d401a675_cmplz_cookiebanners: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_f0d401a675_cmplz_cookies: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_f0d401a675_cmplz_dnsmpd: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_f0d401a675_cmplz_services: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_f0d401a675_cmplz_statistics: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_f0d401a675_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_f0d401a675_comments: Data: 0.08MB + Index: 0.09MB + Engine InnoDB
wp_f0d401a675_csp3_subscribers: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_f0d401a675_layerslider: Data: 1.30MB + Index: 0.00MB + Engine InnoDB
wp_f0d401a675_layerslider_revisions: Data: 9.52MB + Index: 0.00MB + Engine InnoDB
wp_f0d401a675_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_f0d401a675_options: Data: 11.50MB + Index: 0.09MB + Engine InnoDB
wp_f0d401a675_pmxe_exports: Data: 0.08MB + Index: 0.00MB + Engine InnoDB
wp_f0d401a675_pmxe_google_cats: Data: 0.39MB + Index: 0.00MB + Engine InnoDB
wp_f0d401a675_pmxe_posts: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_f0d401a675_pmxe_templates: Data: 0.08MB + Index: 0.00MB + Engine InnoDB
wp_f0d401a675_pmxi_files: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_f0d401a675_pmxi_history: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_f0d401a675_pmxi_images: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
wp_f0d401a675_pmxi_imports: Data: 1.52MB + Index: 0.00MB + Engine InnoDB
wp_f0d401a675_pmxi_posts: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_f0d401a675_pmxi_templates: Data: 0.08MB + Index: 0.00MB + Engine InnoDB
wp_f0d401a675_postmeta: Data: 2.52MB + Index: 1.92MB + Engine InnoDB
wp_f0d401a675_posts: Data: 2.52MB + Index: 0.20MB + Engine InnoDB
wp_f0d401a675_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_f0d401a675_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_f0d401a675_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_f0d401a675_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_f0d401a675_usermeta: Data: 0.08MB + Index: 0.03MB + Engine InnoDB
wp_f0d401a675_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_f0d401a675_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_f0d401a675_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_f0d401a675_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_f0d401a675_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_f0d401a675_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_f0d401a675_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_f0d401a675_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_f0d401a675_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_f0d401a675_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_f0d401a675_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_f0d401a675_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_f0d401a675_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_f0d401a675_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_f0d401a675_wpmailsmtp_tasks_meta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB

### Post Type Counts ###

attachment: 327
atum_inventory_log: 1
cmplz-processing: 1
nav_menu_item: 5
page: 24
post: 1
product: 32
product_variation: 58
revision: 209
seedprod: 1
shop_coupon: 33
shop_order: 91
shop_order_refund: 30

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (21) ###

Query Monitor: by John Blackbourn – 3.6.4
LayerSlider WP: by Kreatura Media – 6.11.2
ATUM Multi-Inventory: by Stock Management Labs™ – 1.4.2
ATUM Inventory Management for WooCommerce: by Stock Management Labs™ – 1.7.9
PublishPress Capabilities: by PublishPress – 1.10.1
CMSMasters Content Composer: by cmsmasters – 2.3.8
CMSMasters Custom Fonts: by cmsmasters – 1.0.1
Coming Soon Page, Maintenance Mode & Landing Pages by SeedProd: by SeedProd – 6.0.8.5
WooCommerce FedEx Shipping Plugin with Print Label: by PluginHive – 4.6.7
jQuery Updater: by Ramoonus – 3.5.1.2
UpdraftPlus - Backup/Restore: by UpdraftPlus.Com
DavidAnderson – 2.16.27.1

WooCommerce AvaTax: by SkyVerge – 1.10.5 (update to version 1.11.0 is available) – Installed version not tested with active version of WooCommerce 4.6.2
WooCommerce Stripe Gateway: by WooCommerce – 4.5.3 – Installed version not tested with active version of WooCommerce 4.6.2
WooCommerce PDF Invoices & Packing Slips: by Ewout Fernhout – 2.7.0
WooCommerce Smart Coupons: by StoreApps – 4.12.2 (update to version 4.13.0 is available)
WooCommerce: by Automattic – 4.6.2 (update to version 4.8.0 is available)
WordPress Importer: by wordpressdotorg – 0.7
WP All Export Pro: by Soflyy – 1.6.2
WP Mail SMTP: by WPForms – 2.5.1
WP Super Cache: by Automattic – 1.7.1
WPS Hide Login: by WPServeur
NicolasKulka
wpformation – 1.5.7

### Dropin Plugins (2) ###

advanced-cache.php: advanced-cache.php
db.php: Query Monitor Database Class

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: ✔

### WC Pages ###

Shop base: #7 - /shop
Cart: #8 - /cart
Checkout: #9 - /checkout
My account: #10 - /my-account
Terms and conditions: #14577 - /terms-of-use

### Theme ###

Name: Sports Store Child
Version: 1.0.0
Author URL: http://cmsmasters.net/
Child Theme: ✔
Parent Theme Name: Sports Store
Parent Theme Version: 1.0.7
Parent Theme Author URL: http://cmsmasters.net/
WooCommerce Support: ✔

### Templates ###

Overrides: -

### Action Scheduler ###

Complete: 185
Oldest: 2020-09-29 21:48:19 -0700
Newest: 2020-09-29 21:48:51 -0700

Pending: 19
Oldest: 2020-09-04 17:02:19 -0700
Newest: 2020-12-03 11:40:15 -0800

`
</details>

**Comment by abhishek-pokhriyal** [comment]
Seems related to #28417

**Comment by pauldowlin** [comment]
Thanks! I obviously didn't search for the correct terms when submitting this.

**Comment by abhishek-pokhriyal** [comment]
If the issue you're facing is the same as the one I linked above then you can expect a fix in WC 4.9.0. You can also test it on the [4.9.0 beta version](https://github.com/woocommerce/woocommerce/releases/tag/4.9.0-beta.1) - on a staging/dev environment of couse

**Comment by pauldowlin** [comment]
Yes, it's exactly the same. Thanks for the heads up!

**Comment by rrennick** [comment]
@pauldowlin Thanks for reporting the issue. It was reported earlier in #28417 and fixed in #28468. The fix will be included in WooCommerce 4.9.0. Closing as a duplicate.
